ParkingLot/src/main/java/services/ParkingSpotManager/ParkingSpotManager.java

```diff
@@ -1,35 +0,0 @@
-package services.ParkingSpotManager;
-
-import models.ParkingSpots.ParkingSpot;
-
-import java.util.List;
-import java.util.Optional;
-
-public abstract class ParkingSpotManager {
-    public List<ParkingSpot> parkingSpotList;
-
-    public ParkingSpotManager(List<ParkingSpot> parkingSpotList) {
-        this.parkingSpotList = parkingSpotList;
-    }
-
-    public Optional<ParkingSpot> findNearestSpot() {
-        // TODO: Taking in entry gate as a parameter and find the nearest spot
-        return parkingSpotList.stream().filter(ParkingSpot::isAvailable).findAny();
-    }
-
-    public boolean occupySpot(ParkingSpot parkingSpot) {
-        Optional<ParkingSpot> optionalParkingSpot = parkingSpotList.stream().filter(spot -> parkingSpot.getId().equals(spot.getId())).findFirst();
```

Comment:
Is two weehler and four wheeler occupied in the same manner? zif yes, why two child classes?

Reply:
Not in current demo. But we might want to override the findNearestSpot for some particular manager.